modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/PrepareServiceImpl.java

```diff
@@ -295,19 +379,36 @@ private CompletableFuture<QueryPlan> prepareExplain(ParsedResult parsedResult, P
                 explicandum
         );
 
-        CompletableFuture<QueryPlan> result;
-        switch (queryType) {
-            case QUERY:
-                result = prepareQuery(newParsedResult, ctx);
-                break;
-            case DML:
-                result = prepareDml(newParsedResult, ctx);
-                break;
-            default:
-                throw new AssertionError("should not get here");
-        }
+        // Validate statement.
+
+        CompletableFuture<ValidStatement> validFut = validateStatement(newParsedResult, ctx);
+
+        // Try optimize to fast
+        CompletableFuture<FastOptimizationResult> plannedOrValidated =
+                validFut.thenApply(stmt -> tryOptimizeFast(stmt, ctx, txContext));
```

Comment:
optimization flow during EXPLAIN and regular flow must share the same path.

Reply:
Fixed.